### PR TITLE
Feat/vmix media reload retrying

### PR DESCRIPTION
**Pull request by jesperstarkar**
* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

Extending https://github.com/nrkno/sofie-timeline-state-resolver/pull/272

* **What is the current behavior?** (You can also link to an open issue here)

* **What is the new behavior (if this is a feature change)?**

* **Other information**:

### CODE REVIEW

**Comment by jesperstarkar** [comment]
Pending testing

File: packages/timeline-state-resolver/src/integrations/vmix/$schemas/options.json [hunk 1]
```diff
@@ -11,10 +11,10 @@
 			"type": "integer",
 			"ui:title": "Port"
 		},
-		"retryInterval": {
+		"pollInterval": {
```
Comment (nytamin) [inline packages/timeline-state-resolver/src/integrations/vmix/$schemas/options.json:14]: Technically this is a breaking change, I'm okay with it though.

File: packages/timeline-state-resolver/src/integrations/vmix/__tests__/vmix.spec.ts [hunk 2]
```diff
@@ -211,6 +212,7 @@ describe('vMix', () => {
 
 		// Time to stop playing
 		await mockTime.advanceTimeToTicks(16050)
+		console.log(JSON.stringify(commandReceiver0.mock.calls, undefined, 4))
```
Comment (nytamin) [inline packages/timeline-state-resolver/src/integrations/vmix/__tests__/vmix.spec.ts:215]: remove this log line

File: packages/timeline-state-resolver/src/integrations/vmix/index.ts [hunk 3]
```diff
@@ -137,31 +131,25 @@ export class VMixDevice extends DeviceWithState<VMixStateExtended, DeviceOptions
...
 		})
 		// this._vmix.on('debug', (...args) => this.emitDebug(...args))
 
 		this._vmix.connect()
 
-		if (typeof options.retryInterval !== 'number' || options.retryInterval > 0) {
-			this._retryTime = options.retryInterval ?? DEFAULT_MEDIA_RETRY_INTERVAL
-			this._retryTimeout = setTimeout(() => this._assertIntendedState(), this._retryTime)
+		if (typeof options.pollInterval !== 'number' || options.pollInterval > 0) {
+			this._pollTime = options.pollInterval ?? DEFAULT_VMIX_POLL_INTERVAL
```
Comment (nytamin) [inline packages/timeline-state-resolver/src/integrations/vmix/index.ts:146]: This logic is pretty hard to follow.

```suggestion
		this._pollTime = options.pollInterval ?? DEFAULT_VMIX_POLL_INTERVAL // options.pollInterval  === 0 disables the polling
		if (this._pollTime  > 0 ) {
			
```

**Review by nytamin — APPROVED** [review]

File: packages/timeline-state-resolver/src/integrations/vmix/index.ts (hunk 3 above)
Comment (jesperstarkar) [reply packages/timeline-state-resolver/src/integrations/vmix/index.ts]: Didn't end up with the exact same implementation.

One side effect of the proposed implementation is that non-numeric numbers will disable all polling, same as <= 0 nums.

If we want this case to fall to the default we need to unwrap this into two IFs.

Comment (jesperstarkar) [reply packages/timeline-state-resolver/src/integrations/vmix/index.ts]: A side-effect of (both the renaming and this change) is that we now have two blocks/implementations of similar-but-different code for the same prop/mechanism between CasparCG and vMix.

**Review by nytamin — APPROVED** [review]
